app/models/choice.rb

```diff
@@ -2,6 +2,8 @@ class Choice < ActiveRecord::Base
   belongs_to(:answer, :inverse_of => :choices, :touch => true)
   belongs_to(:option, :inverse_of => :choices)
 
+  validates :answer, :option, presence: true
```

Comment:
I see. Well just for consistency let's go without the app-level validation. Thanks.